src/assertions.jl

```diff
@@ -0,0 +1,41 @@
+"""
+    Assertion
+
+An Assertion stores settings for checking the validity of a `AbstractArray` or `Tables.table` containing missing values.
+New assertions are expected to subtype `Impute.Assertion` and, at minimum,
+implement the `assert(data::AbstractVector{Union{T, Missing}}, ::<MyAssertion>)` method.
+"""
+abstract type Assertion end
```

Comment:
I don't love this name.

Reply:
Yeah, I'm open to suggestions. Currently, we only have a `Threshold` assertion, but I could see other assertions/checks/tests including checking for `MCAR` or specific missing patterns. If you'd prefer I could remove the abstract type for now and just have the `Threshold` until we have more tests?

Reply:
Rather than needing to decide a name now I opted to make an issue for now https://github.com/invenia/Impute.jl/issues/75. The solution should probably also consider https://github.com/invenia/Impute.jl/issues/74